File: app/src/main/java/com/example/testapplication/pojo/Order.java
package com.example.testapplication.pojo;

import android.os.Parcel;
import android.os.Parcelable;

import io.realm.RealmList;
import io.realm.RealmObject;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = true)
public class Order extends RealmObject implements Parcelable {
    private Client client;
    private long date;
    private String status;
    private boolean sent;
    private double total;
    private RealmList<Item> items;

    public Order() {
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeParcelable(this.client, flags);
        dest.writeLong(this.date);
        dest.writeString(this.status);
        dest.writeByte(this.sent ? (byte) 1 : (byte) 0);
        dest.writeDouble(this.total);
        dest.writeTypedList(this.items);
    }

    protected Order(Parcel in) {
        this.client = in.readParcelable(Client.class.getClassLoader());
        this.date = in.readLong();
        this.status = in.readString();
        this.sent = in.readByte() != 0;
        this.total = in.readDouble();
        this.items = new RealmList<>();
        this.items.addAll(in.createTypedArrayList(Item.CREATOR));
    }

    public static final Creator<Order> CREATOR = new Creator<Order>() {
        @Override
        public Order createFromParcel(Parcel source) {
            return new Order(source);
        }

        @Override
        public Order[] newArray(int size) {
            return new Order[size];
        }
    };
}

File: lib/shared/models/order.dart
import 'dart:convert';

import 'package:orderupv2/shared/models/item.dart';

class Order {
  String to;
  String from;
  String id;
  int date;
  bool forPayment = false;
  List<Item> items;
  String status;
  double total;

  Order({
    this.id,
    this.date,
    this.forPayment,
    this.items,
    this.status,
    this.total,
    this.to,
    this.from,
  });

  Map toJson() {
    return {
      'id' : this.id,
      'date' : this.date,
      'forPayment' : this.forPayment,
      'status' : this.status,
      'items' : this.items != null ? jsonEncode(this.items) : [],
      'total' : this.total,
      'to' : this.to,
      'from' : this.from,
    };
  }
}
